packages/clerk-js/src/ui/elements/DevModeNotice.tsx

```diff
@@ -0,0 +1,55 @@
+import type { ThemableCssProp } from 'ui/styledSystem';
+
+import { useEnvironment } from '../contexts';
+import { Box, Text } from '../customizables';
+
+type DevModeOverlayProps = {
+  gradient?: number;
+};
+
+export const DevModeOverlay = (props: DevModeOverlayProps) => {
+  const { gradient = 60 } = props;
+  const { isDevelopmentOrStaging } = useEnvironment();
+
+  if (!isDevelopmentOrStaging()) {
+    return null;
+  }
+
+  return (
+    <Box
+      sx={t => ({
+        userSelect: 'none',
+        pointerEvents: 'none',
+        inset: 0,
+        position: 'absolute',
+        background: `repeating-linear-gradient(-45deg,${t.colors.$warningAlpha100},${t.colors.$warningAlpha100} 6px,${t.colors.$warningAlpha150} 6px,${t.colors.$warningAlpha150} 12px)`,
+        maskImage: `linear-gradient(transparent ${gradient}%, black)`,
+      })}
+    />
+  );
+};
+
+type DevModeNoticeProps = { sx?: ThemableCssProp };
+export const DevModeNotice = (props: DevModeNoticeProps) => {
+  const { sx } = props;
+  const { isDevelopmentOrStaging } = useEnvironment();
+
+  if (!isDevelopmentOrStaging()) {
+    return null;
+  }
+
+  return (
+    <Text
+      sx={[
+        t => ({
+          color: t.colors.$warning500,
+          fontWeight: t.fontWeights.$semibold,
+          padding: t.space.$1x5,
+        }),
+        sx,
+      ]}
+    >
+      Development mode
```

Comment:
Can/should we add the translation here?

Reply:
Good point, we could add it but not sure what's the use case to have this translated as it will only show up in development. 🤔

Reply:
I guess if someone is developing in a diff language. Non-blocker though!